dotCMS/src/integration-test/java/com/dotcms/util/ImportUtilTest.java

```diff
@@ -1301,6 +1301,130 @@ public void importFile_success_when_lineContainsRelationships(final Relationship
         }
     }
 
+    @Test
+    public void importFile_success_when_lineContainsOneSidedSelfRelationship()
+            throws DotSecurityException, DotDataException, IOException {
+
+        //Creates content types
+        ContentType parentContentType = null;
+        com.dotcms.contenttype.model.field.Field field;
+
+        final int cardinality = RELATIONSHIP_CARDINALITY.MANY_TO_MANY.ordinal();
+
+        try {
+
+            parentContentType = createTestContentType("parentContentType", "parentContentType");
+
+            field = FieldBuilder.builder(RelationshipField.class).name("testRelationship")
+                    .variable("testRelationship")
+                    .contentTypeId(parentContentType.id()).values(String.valueOf(cardinality))
+                    .relationType(parentContentType.variable()).build();
+
+            field = fieldAPI.save(field, user);
+            final Relationship relationship = relationshipAPI.byTypeValue(
+                    parentContentType.variable() + StringPool.PERIOD + field.variable());
+
+            //Creates child contentlet
+            final ContentletDataGen contentletDataGen = new ContentletDataGen(
+                    parentContentType.id());
+            contentletDataGen.languageId(defaultLanguage.getId());
+
+            final Contentlet childContentlet = contentletDataGen
+                    .setProperty(TITLE_FIELD_NAME, "child contentlet")
+                    .setProperty(BODY_FIELD_NAME, "child contentlet").nextPersisted();
```

Comment:
this can be all in one line 
```new ContentletDataGen(
                    parentContentType.id()).languageId(blah).setProperty(blah).nextPersisted();``` (